esphome/components/ethernet/ethernet_component.h

```diff
@@ -10,6 +10,10 @@
 #include "esp_eth.h"
 #include "esp_eth_mac.h"
 #include "esp_netif.h"
+#if ESP_IDF_VERSION_MAJOR >= 5
+  #include "esp_mac.h"
+#else
+#endif
```

Comment:
Based on some testing I did, having this header doesn't seem to hurt in not-IDF 5+

```suggestion
#include "esp_mac.h"
```